ietfdata/datatracker.py

```diff
@@ -82,6 +81,16 @@ def __post_init__(self) -> None:
         assert self.uri.startswith("/api/v1/group/group/")
 
 
+# ---------------------------------------------------------------------------------------------------------------------------------
+# Resource type
+
+@dataclass(frozen=True)
+class Resource:
+    resource_uri : URI
+
+T = TypeVar('T', bound=Resource)
+
+
```

Comment:
I like the `Resource` changes, but they seem unrelated to the cache changes. Would it make sense to split them into a separate PR?

Reply:
They aren't completely unrelated; _retrieve_multi() relies on the object of type T having a resource_uri. That said, the change is probably big enough for its own PR.

Reply:
Okay, makes sense. In that case, maybe leave as-is.

Reply:
OK; will leave it.